app/controllers/users_controller.rb

```diff
@@ -1,3 +1,27 @@
 class UsersController < ApplicationController
-  def new; end
+  def new
+    @user = User.new
+  end
+
+  def show
+    @user = User.find(params[:id])
+  rescue StandardError
+    redirect_to home_path
+  end
+
+  def create
+    @user = User.new user_params
+    if @user.save
+      flash[:success] = t "static_pages.home.welcome"
+      redirect_to user_url @user
```

Comment:
```suggestion
      redirect_to @user
```

hoặc
`redirect_to user_path(id: @user.id)` hoặc `redirect_to user_path @user`

Nên dùng `_path` thay vì `_url` nhé (dùng trong trường hợp này )